### DISCUSSION TITLE
Inference cost - Finn

### DISCUSSION THREAD

**Discussion by KristynaPijackova** [Xilinx/brevitas-radioml-challenge-21#7] [Export & Inference Cost]
Hi all! 

Could you maybe give me a hint on what to do about an import error I get while importing the `inference_cost`?

> ImportError: cannot import name 'inference_cost' from 'finn.util' (unknown location)

For this part in your notebook:
```
from finn.util.inference_cost import inference_cost
import json

inference_cost(export_onnx_path, output_json=cost_dict_path, output_onnx=final_onnx_path,
               preprocess=True, discount_sparsity=True)
```

I tried importing both `!pip install finn` and `!pip install finn-base` and neither worked. Also, when I looked at the API at the websites afterward, they don't seem to include the `inference_cost` either, so now I am kind of clueless on what to try next, any ideas?

**Comment by maltanar** [comment]
Hi @KristynaPijackova , are you using the Docker/Jupyter setup in this sandbox environment, or something custom?

We keep the `inference_cost` function in a special branch of `finn-base` at the moment, you can see how that gets installed in the provided Dockerfile here:

https://github.com/Xilinx/brevitas-radioml-challenge-21/blob/main/sandbox/Dockerfile#L59

`pip install git+https://github.com/Xilinx/finn-base.git@feature/itu_competition_21#egg=finn-base[onnx]`

**Reply by KristynaPijackova** [reply]
Hi Yaman, thank you for pointing that out :) Of course I checked everything but the Docker file... my bad! It works now (and also makes sense why I could not find it in the docs :) ) Thanks!
